src/solver/simulation/adequacy.cpp

```diff
@@ -73,7 +73,11 @@ bool Adequacy::simulationBegin()
         pProblemesHebdo.resize(pNbMaxPerformedYearsInParallel);
         for (uint numSpace = 0; numSpace < pNbMaxPerformedYearsInParallel; numSpace++)
         {
-            SIM_InitialisationProblemeHebdo(study, pProblemesHebdo[numSpace], 168, numSpace);
+            Antares::Data::Area::ScratchMap scratchmap;
+            study.areas.each([&scratchmap, &numSpace](Antares::Data::Area& a) {
+                    scratchmap.try_emplace(&a, &a.scratchpad[numSpace]); });
+
+            SIM_InitialisationProblemeHebdo(study, pProblemesHebdo[numSpace], 168, scratchmap);
```

Comment:
Why not move l.76-78 into `SIM_InitialisationProblemeHebdo` ? After all, these are temporary variables used solely by `SIM_InitialisationProblemeHebdo`.

The last argument could be removed.

Reply:
My goal was to lock the numspace responsibility to economy/adequacy.cpp
I can move it to `SIM_InitialisationProblemeHebdo`